src/scripts/domParsers/videoExtractor.ts

```diff
@@ -0,0 +1,9 @@
+export interface VideoExtractor {
+	// Our 3 Domains all have the concept of a "VideoId" embedded somewhere in the page
+	getVideoIds(pageUrl: string, pageContent: string): string[];
+
+	// We can use the video ideas to generate or retrieve src values that we can embed into our article preview
+	getVideoSrcValues(pageUrl: string, pageContent: string): string[];
```

Comment:
it's a little weird to me that we require passing both `pageUrl` and `pageContent` into this method, but only one or the other will get used, depending on which domain we're on... would it be a lot more complicated to trust the caller to pass in the correct page info, and then just have a `pageInfo` (or similar) parameter?